wkcuber/api/View.py

```diff
@@ -156,6 +161,40 @@ def _check_chunk_size(self, chunk_size):
                 f"The passed parameter 'chunk_size' {chunk_size} must be a multiple of (32, 32, 32)."
             )
 
+    def _handle_compressed_write(self, absolute_offset, data):
+        if (
+            self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4
+            or self.header.block_type == wkw.Header.BLOCK_TYPE_LZ4HC
+        ):
+            # data is compressed
+
+            # calculate aligned bounding box
+            file_bb = np.full(3, self.header.file_len * self.header.block_len)
+            margin_to_top_left = np.array(absolute_offset) % file_bb
+            aligned_offset = np.array(absolute_offset) - margin_to_top_left
+            bottom_right = np.array(absolute_offset) + np.array(list(data.shape)[-3:])
+            margin_to_bottom_right = file_bb - (bottom_right % file_bb)
+            aligned_bottom_right = bottom_right + margin_to_bottom_right
+            aligned_shape = aligned_bottom_right - aligned_offset
+
+            if tuple(aligned_offset) != tuple(absolute_offset) or tuple(
+                aligned_shape
+            ) != tuple(list(data.shape)[-3:]):
```

Comment:
All of those are already tuples.

```suggestion
            if aligned_offset != absolute_offset or aligned_shape != data.shape[-3:]:
```